src/modules/launcher/Plugins/Microsoft.Plugin.Program/Programs/Win32Program.cs

```diff
@@ -885,89 +801,78 @@ private static string GetProgramPathFromRegistrySubKeys(RegistryKey root, string
+                ProgramLogger.Warn($"|Permission denied when trying to load the program from {path}", e, MethodBase.GetCurrentMethod().DeclaringType, path);
 
                 return string.Empty;
             }
         }
 
-        private static Win32Program GetProgramFromPath(string path)
-        {
-            if (string.IsNullOrEmpty(path))
-            {
-                return new Win32Program();
-            }
-
-            path = Environment.ExpandEnvironmentVariables(path);
-
-            if (!File.Exists(path))
-            {
-                return new Win32Program();
-            }
-
-            var entry = CreateWin32Program(path);
-            entry.ExecutableName = Path.GetFileName(path);
-
-            return entry;
-        }
+        private static string ExpandEnvironmentVariables(string path) =>
+            path != null
+                ? Environment.ExpandEnvironmentVariables(path)
+                : null;
 
         // Overriding the object.GetHashCode() function to aid in removing duplicates while adding and removing apps from the concurrent dictionary storage
         public override int GetHashCode()
-        {
-            return RemoveDuplicatesComparer.Default.GetHashCode(this);
-        }
+            => Win32ProgramEqualityComparer.Default.GetHashCode(this);
 
         public override bool Equals(object obj)
-        {
-            return obj is Win32Program win && RemoveDuplicatesComparer.Default.Equals(this, win);
-        }
+            => obj is Win32Program win32Program && Win32ProgramEqualityComparer.Default.Equals(this, win32Program);
 
-        private class RemoveDuplicatesComparer : IEqualityComparer<Win32Program>
+        private class Win32ProgramEqualityComparer : IEqualityComparer<Win32Program>
         {
-            public static readonly RemoveDuplicatesComparer Default = new RemoveDuplicatesComparer();
+            public static readonly Win32ProgramEqualityComparer Default = new Win32ProgramEqualityComparer();
 
             public bool Equals(Win32Program app1, Win32Program app2)
             {
-                if (!string.IsNullOrEmpty(app1.Name) && !string.IsNullOrEmpty(app2.Name)
-                    && !string.IsNullOrEmpty(app1.ExecutableName) && !string.IsNullOrEmpty(app2.ExecutableName)
-                    && !string.IsNullOrEmpty(app1.FullPath) && !string.IsNullOrEmpty(app2.FullPath))
+                if (app1 == null && app2 == null)
                 {
-                    // Using OrdinalIgnoreCase since this is used internally
-                    return app1.Name.Equals(app2.Name, StringComparison.OrdinalIgnoreCase)
-                        && app1.ExecutableName.Equals(app2.ExecutableName, StringComparison.OrdinalIgnoreCase)
-                        && app1.FullPath.Equals(app2.FullPath, StringComparison.OrdinalIgnoreCase);
+                    return true;
                 }
 
-                return false;
+                return app1 != null
```

Comment:
Removed ExecutableName from Equality, as this is always the same as the last part of the FullPath